genetic_algorithms/problems/traveling_salesman_problem/visualization.py

```diff
@@ -0,0 +1,45 @@
+import pygame, sys
+from genetic_algorithms.algorithm_wrappers.algorithm_wrapper import AlgorithmWrapper
+from genetic_algorithms.problems.traveling_salesman_problem.problem_model import TravelingSalesmanModel
+from genetic_algorithms.problems.traveling_salesman_problem.state import TravelingSalesmanState
+
+class Visualization(AlgorithmWrapper):
+    def _perform_side_effects(self, model: TravelingSalesmanModel, state: TravelingSalesmanState):
+        pygame.init()
+        self.screen = pygame.display.set_mode((900, 900))
+        _points = model.points
+        _route = state.route
+        self.find_extreme(_points)
+        _route_points = [(self.scale(_points[idx])) for idx in _route]
+
+        for event in pygame.event.get():
+            if event.type == pygame.QUIT:
+                sys.exit(0)
+
+            elif event.type == pygame.KEYDOWN and event.key == pygame.K_ESCAPE:
+                sys.exit(0)
+
+        self.screen.fill((0, 0, 0))
+        self.draw_points(_route, _points)
+        pygame.draw.lines(self.screen,(0, 150, 0),True,_route_points)
+        pygame.display.flip()
+        pygame.time.delay(500)
+        pygame.display.flip()
```

Comment:
To wyniósłbym do funkcji 
`_draw(route,points)`. 
Wtedy nie będziemy mieszać w funkcji `_perform_side_effects` różnych poziomów abstrakcji (wywołanie funkcji o którym pisałem w komentarzy wyżej oraz operacje na zmiennych o których piszę tutaj.